frontend/src/pages/RegisterStartup.tsx

```diff
@@ -0,0 +1,317 @@
+import { useState } from "react";
+import type { ChangeEvent, FormEvent } from "react";
+import '../css/RegisterStartup.css';
+import opentechlogo from "../img/opentechlogo.png";
+import craftmergelogo from "../img/craftmergelogo.png";
+import craftmergelogoblack from "../img/craftmergelogoblack.png";
+
+type FormState = {
+  name: string;
+  surname: string;
+  email: string;
+  password: string;
+  confirmPassword: string;
+  company: string;
+  registercompany: boolean;
+  startup: boolean;
+  entrepreneur: boolean;
+  legal: boolean;
+};
+
+type ErrorState = Record<string, string>;
+
+export default function RegisterStartup() {
+  const [form, setForm] = useState<FormState>({
+    name: "",
+    surname: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+    company: "",
+    registercompany: false,
+    startup: false,
+    entrepreneur: false,
+    legal: false,
+  });
+
+  const [errors, setErrors] = useState<ErrorState>({});
+  const [status, setStatus] = useState<"idle" | "loading" | "success" | "error">("idle");
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    const { name, value, type, checked, files } = e.target as HTMLInputElement;
+    if (type === "checkbox") {
+      setForm({ ...form, [name]: checked });
+    } else if (type === "file" && files) {
+      setForm({ ...form, [name]: files[0] });
+    } else {
+      setForm({ ...form, [name]: value });
+    }
+  };
+
+  const validate = (): boolean => {
+    const newErrors: Record<string, string> = {};
+    if (!form.email) newErrors.email = "Не ввели електронну пошту";
+    else if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(form.email)) newErrors.email = "Не вірна пошта";
+    if (!form.password) newErrors.password = "Не ввели пароль";
+    else if (form.password.length < 8) newErrors.password = "Пароль дуже малий";
+    if (form.confirmPassword !== form.password)
+      newErrors.confirmPassword = "Не ввели пароль ще раз";
+    if (!form.company) newErrors.company = "Не ввели назву компанії";
+    if (!form.name) newErrors.name = "Не ввели прізвище";
+    if (!form.surname) newErrors.surname = "Не ввели ім’я";
+    if (!form.registercompany && !form.startup) newErrors.startup = "Виберіть кого ви представляєте";
```

Comment:
The validation key for the "who you represent" group uses 'startup' as the error key which is confusing; use a dedicated error key like 'represent' to avoid clashing with the checkbox state key. Change the error assignment and the UI check accordingly.
```
if (!form.registercompany && !form.startup) newErrors.represent = "Виберіть кого ви представляєте";
...
{errors.represent && <p className="error-text">{errors.represent}</p>}
```